models/src/test/scala/coop/rchain/models/rholangn/ParSpec.scala

```diff
@@ -10,29 +10,27 @@ class ParSpec extends AnyFlatSpec with ScalaCheckPropertyChecks with Matchers {
   /** Test hashing and serialization for par
     * @param p1 Par for testing
     * @param p2Opt optional Par (used for testing if necessary to check the correct sorting)
+    * @param eq whether two objects should be equal
     * @return true - if the result of serialization and hashing for both pairs is the same
     */
-  def simpleCheck(p1: ParN, p2Opt: Option[ParN] = None): Boolean = {
+  def simpleCheck(p1: ParN, p2Opt: Option[ParN] = None, eq: Boolean = true): Any = {
     // Serialization and hashing testing
-    val bytes1        = p1.serialized.value
-    val recover1      = Manager.protoDeserialize(bytes1)
-    val res1: Boolean = p1.rhoHash sameElements recover1.rhoHash
+    val bytes1   = p1.serialized.value
+    val recover1 = Manager.protoDeserialize(bytes1)
+    p1.rhoHash.value shouldBe recover1.rhoHash.value
 
     // Testing possibility of calculating the rest of the metadata (without checking correctness)
-    val _ = p1.connectiveUsed || p1.evalRequired || p1.substituteRequired
+    val _ = p1.connectiveUsed.value || p1.evalRequired || p1.substituteRequired
 
     // the correct sorting testing
-    val res2: Boolean = if (p2Opt.isDefined) {
-      val p2     = p2Opt.get
+    p2Opt.map { p2 =>
```

Comment:
When mapping in and _imperative_ style more correct is to call `foreach`  which have a meaning to materialize (evaluate) result. In Option `map` is eagerly evaluated, but with Iterator only foreach will evaluate.